libraries/inverse/hpiFit/signalmodel.cpp

```diff
@@ -0,0 +1,194 @@
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * @brief    signalModel class definition.
+ *
+ */
+
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "hpimodelparameters.h"
+#include "signalmodel.h"
+#include <utils/mnemath.h>
+#include <iostream>
+
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <qmath.h>
+
+//=============================================================================================================
+// EIGEN INCLUDES
+//=============================================================================================================
+
+#include <Eigen/Core>
+
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace INVERSELIB;
+using namespace Eigen;
+
+//=============================================================================================================
+// DEFINE GLOBAL METHODS
+//=============================================================================================================
+
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+SignalModel::SignalModel()
+    : m_iCurrentModelCols(0),
+      m_modelParameters(HpiModelParameters())
+{
+}
+
+//=============================================================================================================
+
+MatrixXd SignalModel::fitData(const HpiModelParameters& hpiModelParameters, const MatrixXd& matData)
+{
+
+    if(checkEmpty(hpiModelParameters)) {
+        MatrixXd matTopo;
+        return matTopo;
+    }
+
+    bool bParametersChanged = m_modelParameters != hpiModelParameters;
```

Comment:
consider marking variables as const if they do not change over the course of the function. Same goes for the rest of this PR.

Reply:
Are you talking about input variables?

Reply:
I mean local variables which are not changed after their initialisation and during their lifecycle 

`const bool bParametersChanged = m_modelParameters != hpiModelParameters;`